### PR TITLE
Implemented slicing semantics for bounded DynamicMaps

### PR DESCRIPTION
**Pull request by jlstevens**
If you start using slices in bounded mode, you have to use slices throughout.

Slicing sets/unsets the dimension soft_ranges which can be used to restrict the explorable region. I'll be documenting this in the new Dynamic Map tutorial.

### CODE REVIEW

**Comment by jlstevens** [comment]
Seems to work fine in the examples I've tried. If it looks ok to you, could you please merge?

File: holoviews/core/spaces.py [hunk 1]
```diff
@@ -551,6 +551,21 @@ def _cross_product(self, tuple_key, cache):
...
+    def _slice_bounded(self, tuple_key):
+        """
+        Slices bounded DynamicMaps by setting the soft_ranges on key dimensions.
+        """
+        cloned = self.clone(self)
+        for i, slc in enumerate(tuple_key):
+            (start, stop) = slc.start, slc.stop
+            if start and start < cloned.kdims[i].range[0]:
+                raise Exception("Requested slice below defined dimension range.")
+            if stop and stop > cloned.kdims[i].range[1]:
```
Comment (philippjfr) [inline holoviews/core/spaces.py]: Make sure you check ``start is not None`` and ``stop is not None`` explicitly unless it'll fall through for zeros.

**Comment by philippjfr** [comment]
Made one comment, looks fine otherwise.

File: holoviews/core/spaces.py (hunk 1 above)
Comment (jlstevens) [inline holoviews/core/spaces.py]: Good point. Fixed in the last commit.

**Comment by philippjfr** [comment]
Looks good, will merge as soon as tests pass.

**Pull request merged**